SUMMARY.md

```diff
@@ -19,6 +19,7 @@
   - [Privacy For Extensibility](./idioms/priv-extend.md)
   - [Easy doc initialization](./idioms/rustdoc-init.md)
   - [Temporary mutability](./idioms/temporary-mutability.md)
+  - [Fallible closure](./idioms/fallible-closure.md)
```

Comment:
We might find a better headline for the summary to let this idiom be found more easily.

Reply:
I agree, this often comes up when people want to avoid "arrow code" with let guards or "match hell", maybe something around that ?